src/overlays/actors/ovl_En_Horse_Normal/z_en_horse_normal.c

```diff
@@ -21,45 +48,635 @@ const ActorInit En_Horse_Normal_InitVars = {
+        func_800A663C(globalCtx, &this->skin, &D_06009FAC, &D_06004580);
+        SkelAnime_ChangeAnimDefaultStop(&this->skin.skelAnime, sAnimations[this->animationIdx]);
+        if ((this->actor.posRot.pos.x == -730.0f && this->actor.posRot.pos.y == 0.0f &&
+             this->actor.posRot.pos.z == -1100.0f) ||
+            (this->actor.posRot.pos.x == 880.0f && this->actor.posRot.pos.y == 0.0f &&
+             this->actor.posRot.pos.z == -1170.0f)) {
+            func_80A6C6B0(this);
+            return;
+        }
+    } else if (globalCtx->sceneNum == SCENE_MALON_STABLE) {
+        if (!gSaveContext.nightFlag) {
+            Actor_Kill(&this->actor);
+        } else {
+            func_800A663C(globalCtx, &this->skin, &D_06009FAC, &D_06004580);
+            SkelAnime_ChangeAnimDefaultStop(&this->skin.skelAnime, sAnimations[this->animationIdx]);
+            func_80A6C6B0(this);
+        }
+        return;
+    } else if (globalCtx->sceneNum == SCENE_SPOT12) {
+        if (this->actor.posRot.pos.x == 3707.0f && this->actor.posRot.pos.y == 1413.0f &&
+            this->actor.posRot.pos.z == -665.0f) {
+            func_800A663C(globalCtx, &this->skin, &D_06009FAC, &D_06004580);
+            SkelAnime_ChangeAnimDefaultStop(&this->skin.skelAnime, sAnimations[this->animationIdx]);
+            func_80A6C4CC(this);
+            return;
+        }
+        func_800A663C(globalCtx, &this->skin, &D_06009FAC, &D_06004580);
+        SkelAnime_ChangeAnimDefaultStop(&this->skin.skelAnime, sAnimations[this->animationIdx]);
+    } else {
+        func_800A663C(globalCtx, &this->skin, &D_06009FAC, &D_06004580);
+        SkelAnime_ChangeAnimDefaultStop(&this->skin.skelAnime, sAnimations[this->animationIdx]);
+    }
+    if ((this->actor.params & 0xF0) == 0x10 && (this->actor.params & 0xF) != 0xF) {
+        func_80A6B91C(this, globalCtx);
+    } else {
+        func_80A6BC48(this);
+    }
+}
+
+void EnHorseNormal_Destroy(Actor* thisx, GlobalContext* globalCtx) {
+    EnHorseNormal* this = THIS;
+
+    func_800A6888(globalCtx, &this->skin);
+    Collider_DestroyCylinder(globalCtx, &this->bodyCollider);
+    Collider_DestroyCylinder(globalCtx, &this->cloneCollider);
+    Collider_DestroyJntSph(globalCtx, &this->headCollider);
+}
+
+void func_80A6B91C(EnHorseNormal* this, GlobalContext* globalCtx) {
+    this->actor.flags |= 0x10;
+    this->type = 4;
+    this->animationIdx = 6;
+    this->waypoint = 0;
+    this->actor.speedXZ = 7.0f;
+    SkelAnime_ChangeAnim(&this->skin.skelAnime, sAnimations[this->animationIdx], func_80A6B30C(this), 0.0f,
+                         SkelAnime_GetFrameCount(&sAnimations[this->animationIdx]->genericHeader), 2, 0.0f);
+}
+
+void EnHorseNormal_FollowPath(EnHorseNormal* this, GlobalContext* globalCtx) {
+    Path* path = &globalCtx->setupPathList[this->actor.params & 0xF];
+    Vec3s* pointPos = SEGMENTED_TO_VIRTUAL(path->points);
+    f32 dx;
+    f32 dz;
+    s32 pad;
+
+    pointPos += this->waypoint;
```

Comment:
```suggestion
    Vec3s* pointPos = &SEGMENTED_TO_VIRTUAL(path->points)[this->waypoint];
    f32 dx;
    f32 dz;
    s32 pad;

```
if this makes it less readable please ignore, my reasoning is that it avoids pointPos being used as a temp but no big deal (may also not match?)

Reply:
It doesn't match.